pkg/scheduler/apis/config/v1beta2/default_plugins.go

```diff
@@ -170,16 +165,20 @@ func mergePluginSet(defaultPluginSet, customPluginSet v1beta2.PluginSet) v1beta2
 			}
 			// The default plugin is explicitly re-configured, update the default plugin accordingly.
 			if customPlugin, ok := enabledCustomPlugins[defaultEnabledPlugin.Name]; ok {
-				klog.InfoS("Defaut plugin is explicitly re-configured and is overriding.", "plugin", defaultEnabledPlugin.Name)
+				klog.InfoS("Default plugin is explicitly re-configured and is overriding.", "plugin", defaultEnabledPlugin.Name)
 				// update the default plugin in place to preserve order
-				defaultEnabledPlugin = customPlugin.plugin
-				// kick the plugin from the enabled list of custom plugins
-				customPluginSet.Enabled = append(customPluginSet.Enabled[:customPlugin.index], customPluginSet.Enabled[customPlugin.index+1:]...)
+				defaultEnabledPlugin = customPlugin
+				delete(enabledCustomPlugins, defaultEnabledPlugin.Name)
 			}
 			enabledPlugins = append(enabledPlugins, defaultEnabledPlugin)
 		}
 	}
 
-	enabledPlugins = append(enabledPlugins, customPluginSet.Enabled...)
+	// append all the custom plugins which are not defined as default plugins.
+	for _, plugin := range customPluginSet.Enabled {
+		if _, ok := enabledCustomPlugins[plugin.Name]; ok {
```

Comment:
Isn't this getting rid of repeated elements (if they are also default)?

Reply:
I think so, here are three choices, 
- If there are repeated elements, accept the first one, and ignore others, this is what's this PR does, but we need to add some note somewhere.
- throw an exception here, and say the repeated elements are not allowed, which align with the logic when the framework are initialized.
- try to save the index of custom plugin somewhere and compare the index when appending the left custom plugins, so it will fail later during the framework instantiation (maybe a little dirty to achieve this.) 

second choices looks easier, what do you think?

Reply:
I prefer the third option. Defaulting shouldn't validate.

Reply:
pls refer to the testcase "RepeatedCustomPlugin", the repeated custom plugin will be kept.